lib/lf/base/comm.h

```diff
@@ -0,0 +1,238 @@
+
+namespace input {
+
+extern int kArgc;
+extern char** kArgv;
+extern std::string kConfigFile;
+extern po::variables_map kVM;
+extern po::options_description kDesc;
+
+extern void Init(int argc, char** argv, std::string file);
+extern void Init(int argc, char** argv);
+extern void Init(std::string file);
+extern void Init();
+
+extern po::options_description_easy_init Add();
+extern void Add(const std::string& name, const std::string& comment);
+template <typename T>
+void Add(const std::string& name, const std::string& comment);
+template <typename T>
+void Add(const std::string& name, const std::string& comment, const T& def);
+template <class A>
+void AddCtrl(const std::string& name, A& class_instance,
+             const std::string& comment = "");
+template <typename T>
+void AddSetter(const std::string& name, T& value,
+               const std::string& comment = "");
+
+template <typename T>
+T Get(const std::string& name);
+template <typename T>
+T Get(const std::string& name, const T& alt);
+
+extern bool Help();
+
+extern bool IsSet(const std::string& name);
+
+extern void ParseCommandLine(const int argc = 0, const char** argv = nullptr);
+
+extern bool ParseFile(const std::string& file = "");
+
+/**
+ * @brief Add possible input for variable called `name` with description
+ *        `comment`.
+ * @param name The name of the variable.
+ * @param comment Description of the variable.
+ */
+template <typename T>
+void Add(const std::string& name, const std::string& comment) {
+  kDesc.add_options()(name.c_str(), po::value<T>(), comment.c_str());
+}
+
+/**
+ * @brief Add possible input for variable called `name` with description
+ *        `comment` with default value `def`.
+ * @param name The name of the variable.
+ * @param comment Description of the variable.
+ * @param def T The default value.
+ */
+template <typename T>
+void Add(const std::string& name, const std::string& comment, const T& def) {
+  kDesc.add_options()(name.c_str(), po::value<T>()->default_value(def),
+                      comment.c_str());
+}
+
+template <class A>
+std::function<void(unsigned int)> SetCtrl(A& a) {
+  std::function<void(unsigned int)> lambda = [&a](unsigned int c) {
+    a.output_ctrl_ = c;
+  };
+  return lambda;
+}
+
+/**
+ * @brief Possibility of setting the public member `ctrl` of the
+ *        `class_instance`. Option is called `name` with description `comment`
+ * @param name The name of the option.
+ * @param class_instance Instance of the class wher to change member value.
+ * @param comment (optional) Description of the option.
+ */
+template <class A>
+void AddCtrl(const std::string& name, A& class_instance,
+             const std::string& comment) {
+  // Why not solve it like:
+  //   void AddCtrl(const string& name, const T& val, const string& comment);
+  // Call it like:
+  //   AddCtrl("ctrl", MyClass.PublicVar, "Some comment");
+  //   AddCtrl("ctrl", MyClass::StaticVar, "Some comment");
+  // ?
+  kDesc.add_options()(
```

Comment:
Exactly, I think this would be the better approach. I don't think we have to provide specialized methods for setting members named `output_ctrl_` in a class. I would actually get rid of `AddCtrl` completly and just use `AddSetter` which is a much better implementation...

Reply:
done :+1: